File: 234.palindrome-linked-list.java
import java.util.Stack;

/*
 * @lc app=leetcode id=234 lang=java
 *
 * [234] Palindrome Linked List
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    int sum=0;

    public boolean isPalindrome(ListNode head) {



        
    }
    public ListNode middleFinder(ListNode head){

        ListNode fast = head;
        ListNode slow = head;
        Stack<Integer> palindrome = new Stack<Integer>();
int i =1;

while (fast != null && fast.next != null) {
    fast = fast.next.next;
    slow = slow.next;
}
if (fast != null) { // odd nodes: let right half smaller
    slow = slow.next;
}


    



    }
}
// @lc code=end

File: 234.palindrome-linked-list.py
#
# @lc app=leetcode id=234 lang=python3
#
# [234] Palindrome Linked List
#

# @lc code=start
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
from typing import Optional


class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        if not head or not head.next:
            return True
        stack = []
        while head:
            if stack and stack[-1] == head.val:
                stack.pop()
            else:
                stack.append(head.val)
            head = head.next
        print(stack)
        return not stack
        
# @lc code=end

